.circleci/config.yml

```diff
@@ -36,22 +36,31 @@ jobs:
     machine:
       image: ubuntu-2204:current
     steps:
-      - run: git clone https://github.com/heroku/<< parameters.language >>-getting-started.git getting_started
+      - run: git clone https://github.com/heroku/<< parameters.language >>-getting-started.git getting-started
       - pack/install-pack:
           version: 0.26.0
       - attach_workspace:
           at: /tmp/workspace
       - run: docker pull heroku/heroku:<< parameters.stack-tag>>-cnb
       - run: zstd -dc /tmp/workspace/<< parameters.builder-dir >>.tar.zst | docker load
       - run: pack config trusted-builders add << parameters.builder-dir >>
-      - run: pack build pack-getting-started --path getting_started --builder << parameters.builder-dir >> --pull-policy never
-      - run: docker run --name getting-started -d -e PORT=8080 pack-getting-started
-      - run: sleep 5
-      - run: docker exec getting-started true 2>/dev/null || (echo not running && docker logs getting-started && exit 1)
-  test-example:
+      - run: pack build getting-started --path getting-started --builder << parameters.builder-dir >> --pull-policy never
+      - run: docker run --name getting-started --detach -p 8080:8080 --env PORT=8080 getting-started
+      - run:
+          name: Test web server response
+          command: |
+            if curl -sSf --retry 5 --retry-delay 1 --retry-all-errors --connect-timeout 3 http://localhost:8080 -o response.txt; then
+              echo "Successful response from server"
+            else
+              echo "Server did not respond successfully"
+              docker logs getting-started
+              [[ -f response.txt ]] && cat response.txt
+              exit 1
+            fi
+  test-example-function:
```

Comment:
I've renamed this job type from `test-example` to `test-example-function`, because the `x-health-check` handling in the `Test function response` step is functions specific.

If we ever have additional fixtures under `examples/` that are no functions, we can add a new job type at that point to test them separately.